src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -138,8 +139,41 @@ internal static bool TryGetBestOverload(CheckTypesContext context, IErrorContain
 
                 var localWarnings = new LimitedSeverityErrorContainer(errors, DocumentErrorSeverity.Warning);
 
+                var checkTypeErrors = new ErrorContainer();
+
                 // Typecheck the invocation and infer the return type.
-                typeCheckSucceeded = maybeFunc.CheckTypes(context, args, argTypes, localWarnings, out returnType, out nodeToCoercedTypeMap);
+                typeCheckSucceeded = maybeFunc.CheckTypes(context, args, argTypes, checkTypeErrors, out returnType, out nodeToCoercedTypeMap);
+
+                if (!typeCheckSucceeded && checkTypeErrors.HasErrors() &&
+                    checkTypeErrors.GetErrors().All(error => context.NameResolver.Lookup(new DName(error.Tok.ToString()), out NameLookupInfo nameLookup) && nameLookup.Type.IsDeferred))
```

Comment:
Why are we iterating the error container/calling Lookup?
We have the types of the args in `argTypes`